Add deployment guide of AWS EC2 (#550)
kestra-io/docs: content/docs/09.administrator-guide/02.deployment/06.ec2.md

## Step 3: Allow external traffic

Kestra is now running and the Kestra server exposes traffic on the 8080 port. To be able to connect through our web browser we have to update inbound traffic rules in our EC2 security group.

To do so, go to the EC2 console, select Security Group and choose the security group attached to your EC2 instance.

Add a new inbound rule to open access to the 8080 port. 

If you want to only allow traffic coming from your IP you can set the source to your own IP. If you want to make it open to the whole Internet you can leave 0.0.0.0. Note that if you didn't set basic authentication in the previous step, everyone could access your Kestra instance without any access restriction.

![ec2 inbound rules](/docs/administrator-guide/deployment/aws-ec2/ec2_security_group_port_inbound_rules.png)

You should now have access to your Kestra instance and create, edit and run Flow !

## Step 4: Use AWS RDS for proper database setup

This first installation relies on a Postgre database running alongside the Kestra server - on the EC2 instance (see the postgres service running thanks to the docker-compose).

> For a production-grade installation, you would prefer to use a proper database setup. To do, so we’ll create an AWS RDS database.

Reviewer: ```suggestion
For a simple PoC, you can keep that Postgres database running in Docker.

However, for a production-grade installation, we recommend a managed database service such as AWS RDS.
```

**Create a AWS RDS database**

> 1. Go on the [RDS console](https://eu-north-1.console.aws.amazon.com/rds/home).

Reviewer: ```suggestion
1. Go to the [RDS console](https://eu-north-1.console.aws.amazon.com/rds/home).
```

> 2. Create a database and choose Postgres (Kestra also supports MySQL)

Reviewer: ```suggestion
2. Create a database and choose Postgres (Kestra also supports MySQL, but Postgres is recommended)
```

> 3. Set a proper username and password.

Reviewer: ```suggestion
3. Set a username and password.
```

4. On the connectivity configuration choose “Connect to an EC2 compute resource” and choose your EC2 instance.
5. Also select the existing DB subnet group and existing VPC security group and choose the one attached to your EC2 instance. 
5. Fine tune instance class, storage  type to avoid import AWS costs. For a first step a small Postgres instance is enough
6. Hit create and wait for completion

![rds setup](/docs/administrator-guide/deployment/aws-ec2/rds_setup1.png)

![rds connectivity](/docs/administrator-guide/deployment/aws-ec2/rds_setup2.png)

**Create kestra database**

> Before attaching our Kestra server to our new database backend, we have to initialize the database with a base schema:

Reviewer: ```suggestion
Before attaching your Kestra server to our new database backend, initialize the database with a base schema as follows:
```

1. Connect to your EC2 instance with ssh
2. Install a PostgresSQL client: `sudo apt-get install postgresql-client`
3. Create the kestra database: `createdb -h <your-rds-url-endpoint> -U <your-username> -p 5432 kestra`

**Update Kestra configuration**

> In the docker-compose configuration, edit the datasources property of the kestra service like below:

Reviewer: ```suggestion
In the docker-compose configuration, edit the datasources property of the kestra service in the following way:
```

```yaml
datasources:
    postgres:
    url: jdbc:postgresql://<your-rds-url-endpoint>:5432/kestra
    driverClassName: org.postgresql.Driver
    username: <your-username>
    password: <your-password>
```

> As we have moved to RDS service, we don't need the docker-compose postgres service anymore. You can remove it.

Reviewer: ```suggestion
Because you now use RDS service, you don't need the postgres service anymore. Remove it from the `docker-compose.yml` file.
```

> You can then restart the docker services with `docker compose restart`.

Reviewer: ```suggestion
In order for the changes to take effect, restart the docker services with `docker compose restart` or `docker compose up -d`.
```

## Step 5: Use AWS S3 for storage

> Kestra also uses by default the local file system for internal storage. To better persist executions and files we’ll set up AWS S3 as the internal storage backend.

Reviewer: ```suggestion
By default, internal storage is implemented using the local file system. This section will guide you on how to change the storage backend to S3 to ensure a more reliable, durable and scalable storage.
```

1. Go to the S3 console and create a bucket.
2. Go to IAM and create a new User Group with AWS full access.
3. Create a new user and attach it to the user group.
4. For the new user, go to Security Credentials and create an access key. Choose “Application running on an AWS compute service” and retrieve the access and secret keys 
5. Edit Kestra storage configuration

>     ```yaml
>     kestra:
>     storage:
>         type: s3
>         s3:
>         accessKey: "<your-aws-access-key-id>"
>         secretKey: "<your-aws-secret-access-key>"
>         region: "<your-aws-region>"
>         bucket: "<your-s3-bucket-name>"
>     ```

Reviewer: ```suggestion
```yaml
kestra:
  storage:
      type: s3
      s3:
      accessKey: "<your-aws-access-key-id>"
      secretKey: "<your-aws-secret-access-key>"
      region: "<your-aws-region>"
      bucket: "<your-s3-bucket-name>"
```
```
